api/task_lifecycle_client.go

```diff
@@ -81,6 +88,42 @@ func (c *TaskLifecycleClient) Scheme() string {
 	return fmt.Sprintf(c.scheme)
 }
 
+// CreateTask takes a task request and run option and sends this information to the client. It then returns
+// a task response object and any errors to the caller.
+func (c *TaskLifecycleClient) CreateTask(ctx context.Context, runOption string, req TaskRequest) (TaskResponse, error) {
+	var run oapigen.CreateTaskParamsRun
+	switch runOption {
+	case RunOptionInspect:
+		run = RunOptionInspect
+	case RunOptionNow:
+		run = RunOptionNow
+	default:
+		err := errors.New("invalid run option provided")
+		return TaskResponse{}, err
+	}
+
+	resp, err := c.Client.CreateTask(ctx, &oapigen.CreateTaskParams{Run: &run}, oapigen.CreateTaskJSONRequestBody(req))
+	if resp != nil {
+		defer resp.Body.Close()
+	}
+	if err != nil {
+		err = fmt.Errorf("error creating task %s, %w", req.Name, err)
```

Comment:
What do these errors end up looking like with the wrapper? I'm cautious that this is redundant for the client and would be the last bubbled up error with very similar wraps.

Reply:
I just threw in a dummy error to get this to trip, this is what it looks like

```
==> Inspecting changes to resource if creating task 'api-task'...

    Generating plan that Consul Terraform Sync will use Terraform to execute

==> Error: unable to generate plan for 'api-task'
    error creating task api-task, this is an error
```

Reply:
maybe this doesn't need to be wrapped, it does seem a bit redundant

Reply:
I've seen the wrapping multiple times, so i don't think the dummy error is representative of what the nested ones would surface.

Reply:
Ah I misunderstood your first statement then, you're thinking that the server might bubble up a similar `fmt.Errorf("error creating task %s, %w", req.Name, err)` so we might get the same error written multiple times.

Well then, maybe `task client error` adds the context without being redundant?

Reply:
hmm but then there isn't really an error with the client... I think it would be better to just return the error without wrapping in this case

Reply:
Just manually going through it, I think this is what the errors end up looking like:

```
==> Inspecting changes to resource if creating task 'api-task'...

    Generating plan that Consul Terraform Sync will use Terraform to execute

==> Error: unable to generate plan for 'api-task'
    error creating task api-task, request returned 400 status code with error: error decoding the request: some error

==> Error: unable to generate plan for 'api-task'
    error creating task api-task, request returned 400 status code with error:  task with name api-task already exists

==> Error: unable to generate plan for 'api-task'
    error creating task api-task, request returned 400 status code with error:  error with task configuration: some error

==> Error: unable to generate plan for 'api-task'
    error creating task api-task, request returned 400 status code with error:  this is an error
```

Reply:
I don't think it's any less clear without the wrapping

```
==> Error: unable to generate plan for 'api-task'
        request returned 400 status code with error: error decoding the request: some error

==> Error: unable to generate plan for 'api-task'
        request returned 400 status code with error:  task with name api-task already exists

==> Error: unable to generate plan for 'api-task'
        request returned 400 status code with error:  error with task configuration: some error

==> Error: unable to generate plan for 'api-task'
        request returned 400 status code with error:  this is an error
```

Reply:
Ah ok, I wasn't sure how many and whether the other error wraps would bubble up. Looks like this path is clear, but sounds good that it's clear with and without it.